src/odemis/driver/xt_client.py

```diff
@@ -2374,58 +2374,56 @@ def _moveTo(self, future: futures.Future, pos: Dict[str, float], rel: bool = Fal
                 # (1) not identical to the starting position and (2) not too far from the target position.
 
-                # The stage is not moving anymore, but we still want to wait until the position has been updated
-                # TODO: base the timeout on the stage movement time estimation (est. *10)
-                timeout = 10  # s
-                expected_end_time = time.time() + timeout  # s
-                check_pos = True
-
-                # TODO: Include rotation and tilt to check stage movement to target position. Test it on a hardware.
-                # The below check to skip the check on stage movement to target position if it includes r ot t
-                # is temporary. It will be removed once the behavior is extended to all the axes and confirmed to work
-                # on a hardware.
-                if 'r' in pos or 't' in pos:
-                    logging.debug("Position requested to move includes t (rx/tilt) or r (rz/rotation). "
-                                  "Reported position accuracy is not checked.")
-                    check_pos = False
-
                 # Get the target position in absolute coordinates
                 if rel:
                     target_pos = {ax: val + pos[ax] for ax, val in orig_pos.items()}
                 else:
                     target_pos = pos
 
+                # The stage is not moving anymore, but we still want to wait until the position has been updated
+                # TODO: base the timeout on the stage movement time estimation (est. *10)
+                timeout = 10  # s
+                expected_end_time = time.time() + timeout  # s
+                check_pos = True
+                fraction_tolal_move = 0.01  # percentage of tolerance from the requested stage movement
+                tol_linear = 1e-06  # in m, minimum tolerance
+                tol_rotational = 0.00436  # in radians (0.25 degrees), minimum tolerance
+
+                # Update rotational and linear tolerances according to the magnitude of requested change
+                linear_axes_to_check = {"x", "y", "z"}.intersection(target_pos.keys())
+                rotational_axes_to_check = {"t", "r"}.intersection(target_pos.keys())
+                current_pos = self.parent.get_stage_position()
+                movement_req = {ax: fraction_tolal_move * abs(target_pos[ax] - current_pos[ax]) for ax in
+                                target_pos.keys()}
+                if linear_axes_to_check:
+                    movement_req_linear = [movement_req[ax] for ax in linear_axes_to_check]
+                    tol_linear = max(min(movement_req_linear), 1e-06)  # m
+                if rotational_axes_to_check:
+                    movement_req_rotational = [movement_req[ax] for ax in rotational_axes_to_check]
+                    tol_rotational = max(min(movement_req_rotational), 0.004363323129985824)  # 0.25 degrees
+
```

Comment:
Remove magic numbers, and use initial constants instead
```suggestion
                linear_axes_to_check = {"x", "y", "z"}.intersection(target_pos.keys())
                rotational_axes_to_check = {"t", "r"}.intersection(target_pos.keys())
                current_pos = self.parent.get_stage_position()
                movement_req = {ax: FRACTION_TOTAL_MOVE * abs(target_pos[ax] - current_pos[ax]) for ax in
                                target_pos.keys()}
                if linear_axes_to_check:
                    movement_req_linear = [movement_req[ax] for ax in linear_axes_to_check]
                    TOL_LINEAR = max(min(movement_req_linear), TOL_LINEAR)  # m
                if rotational_axes_to_check:
                    movement_req_rotational = [movement_req[ax] for ax in rotational_axes_to_check]
                    TOL_ROTATION = max(min(movement_req_rotational), TOL_ROTATION)  # 0.25 degrees

```